File: app/src/main/java/ua/com/foxminded/locationtrackera/ui/tracker/dialog/TrackerDialogContract.java
package ua.com.foxminded.locationtrackera.ui.tracker.dialog;

import ua.com.foxminded.locationtrackera.mvi.FragmentContract;
import ua.com.foxminded.locationtrackera.ui.tracker.dialog.state.TrackerDialogScreenEffect;
import ua.com.foxminded.locationtrackera.ui.tracker.dialog.state.TrackerDialogScreenState;

public class TrackerDialogContract {

    public interface ViewModel extends FragmentContract.ViewModel<TrackerDialogScreenState, TrackerDialogScreenEffect> {
    }

    public interface View extends FragmentContract.View {
    }

    public interface Host extends FragmentContract.Host {
    }
}

File: app/src/main/java/ua/com/foxminded/locationtrackera/ui/tracker/dialog/TrackerDialogContract.kt
package ua.com.foxminded.locationtrackera.ui.tracker.dialog

import ua.com.foxminded.locationtrackera.mvi.fragments.FragmentContract
import ua.com.foxminded.locationtrackera.ui.tracker.dialog.state.TrackerDialogScreenEffect
import ua.com.foxminded.locationtrackera.ui.tracker.dialog.state.TrackerDialogScreenState

class TrackerDialogContract {
    interface ViewModel : FragmentContract.ViewModel<
            TrackerDialogScreenState,
            TrackerDialogScreenEffect>

    interface View : FragmentContract.View
    interface Host : FragmentContract.Host
}
